tests/src/test/java/org/mozilla/javascript/tests/JavaAdapterInvokeTest.java

```diff
@@ -0,0 +1,128 @@
+            this.adapter = adapter;
+        }
+
+        public int doIt(int i) {
+            return this.adapter.m1(i) + this.adapter.m2();
+        }
+    }
+
+    /**
+     * This test creates a new {@link AdapterClass} and passes a javascript object with two
+     * functions (m1,m2) to the constructor.
+     *
+     * <p>It is expected, that rhino creates an {@link AdapterInterface}, that delegates the calls
+     * to m1/m2 back to the javascript implementation.
+     */
+    @Test
+    public void testInvoke() {
+        String testCode =
+                "'use strict'\n"
+                        + "var impl = {"
+                        + "  m1: function(i) { return i + 1 },\n"
+                        + "  m2: function() { return 7 }\n"
+                        + "}\n"
+                        + "adapter = new Packages."
+                        + AdapterClass.class.getName()
+                        + "(impl)\n"
+                        + "adapter.doIt(42)";
+        Utils.runWithAllOptimizationLevels(
+                cx -> {
+                    final Scriptable scope = cx.initStandardObjects();
+                    Number result = (Number) cx.evaluateString(scope, testCode, "", 1, null);
+                    Assert.assertEquals(50, result.intValue());
+                    return null;
+                });
+    }
+
+    /** Similar to {@link #testInvoke()}, but we use a javascript object, created from prototype. */
+    @Test
+    public void testInvokeWithPrototype() {
+        String testCode =
+                "'use strict'\n"
+                        + "function Obj() {}\n"
+                        + "Obj.prototype.m1 = function(i) { return i + 1 }\n"
+                        + "Obj.prototype.m2 = function() { return 7 }\n"
+                        + "var impl = new Obj()\n"
+                        + "adapter = new Packages."
+                        + AdapterClass.class.getName()
+                        + "(impl)\n"
+                        + "adapter.doIt(42)";
+
+        Utils.runWithAllOptimizationLevels(
+                cx -> {
+                    final Scriptable scope = cx.initStandardObjects();
+                    Number result = (Number) cx.evaluateString(scope, testCode, "", 1, null);
+                    Assert.assertEquals(50, result.intValue());
+                    return null;
+                });
+    }
+
+    /**
+     * Test uses some value in the method that are either constructed in ctor or in prototype. This
+     * test fails before #1453
+     */
+    @Test
+    public void testInvokeWithPrototypeAndObjs() {
+        String testCode =
+                "'use strict'\n"
+                        + "function Obj() { this.myObj = {one: 1} }\n"
+                        + "Obj.prototype.seven = 7\n"
+                        + "Obj.prototype.m1 = function(i) { return i + this.myObj.one }\n"
+                        + "Obj.prototype.m2 = function() { return this.seven }\n"
```

Comment:
This test shows the problem. `this.myObj` and `this.seven` were undefined, when the js object is passed in an `InterfaceAdapter`
The other tests are all passing without the fix and are for completeness, not to break anything else